rql/parser.go

```diff
@@ -0,0 +1,201 @@
+package rql
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/raystack/salt/utils"
+)
+
+var validNumberOperations = []string{"eq", "neq", "gt", "gte", "lte"}
+var validStringOperations = []string{"eq", "neq", "like"}
+var validBoolOperations = []string{"eq", "neq"}
+var validDatetimeOperations = []string{"eq", "neq", "gt", "gt", "gte", "lte"}
+var validSortOrder = []string{"asc", "desc"}
+
+const TAG = "rql"
+
+type Query struct {
+	Filters []Filter `json:"filters"`
+	GroupBy []string `json:"group_by"`
+	Offset  int      `json:"offset"`
+	Limit   int      `json:"limit"`
+	Search  string   `json:"search"`
+	Sort    []Sort   `json:"sort"`
+}
+
+type Filter struct {
+	Name     string `json:"name"`
+	Operator string `json:"operator"`
+	Type     string `json:"type"`
```

Comment:
Since we are not exposing the the type field outside, any reason to keep it public and also assigning the json tag to it. 
Context: Ideally, we would be using the Query and Filter structs as our payload type so, we shouldn't include `json` tags if we are not supporting it.

Reply:
yeah makes sense

Reply:
made it `dataType` since `type` is a Golang reserved literal.